lib/monetary-amount.js

```diff
@@ -51,14 +109,15 @@ class MonetaryAmount {
      * @returns {MonetaryAmount|null} A new instance or null if parsing failure
      */
     static from(json) {
-        if (json) {
-            if (typeof json.amount !== 'undefined'
-                && typeof json.currency !== 'undefined'
-                && typeof json.currency.ct !== 'undefined'
-                && typeof json.currency.id !== 'undefined'
-            ) return new MonetaryAmount(json.amount, json.currency.ct, json.currency.id);
+        try {
+            if ((typeof json !== 'object') || (typeof json.currency !== 'object'))
+                throw 0;
+
+            return new MonetaryAmount(json.amount, json.currency.ct, json.currency.id);
+        }
+        catch (err) {
+            return null;
```

Comment:
This is not how the factory function should be implemented. It should be implemented without the use of internal exceptions!

Reply:
Done

Reply:
Thank you for the clarification on the phone.
Removed parsing from the constructor by restricting input arguments to types that do not require parsing, changed the game.